modules/OFConnectionManager2/module/src/ofconnectionmanager.c

```diff
@@ -1118,8 +1119,13 @@ ind_cxn_set_aux_cxns(connection_t *main_cxn, uint32_t num_aux)
             return 1;
         }
     } else {
+        controller_t *controller = main_cxn->controller;
         for (idx = main_cxn->controller->num_aux+1; idx <= num_aux; ++idx) {
             if (controller_add_cxn(main_cxn->controller, idx) == NULL) {
+                while (--idx > controller->num_aux) {
+                    controller_stop_cxn(controller->cxns[idx]);
+                    controller->cxns[idx] = NULL;
+                }
```

Comment:
just trying to avoid adding more duplicate code...

i'm also wondering what the behavior should actually be, since the previous behavior is that we have 0 or more new aux connections, and now the behavior is "if we can't honor the controller request, we're going to kill the controller".